packages/ui-components/src/lib/form/Form.tsx

```diff
@@ -25,24 +26,56 @@ export interface UnsavedChangesConfig {
   modalProps?: UnsavedChangesModalProps;
 }
 
-export interface FormProps<T> extends FormikConfig<T> {
+export interface PersistValuesConfig<T> {
+  persistFunction: (values: T) => void;
+  debounce?: number;
+}
+
+export interface FormProps<T> extends Omit<FormikConfig<T>, 'onSubmit'> {
   className?: string;
   confirmUnsavedChanges?: boolean;
+  persistValuesConfig?: PersistValuesConfig<T>;
   unsavedChangesConfig?: UnsavedChangesConfig;
   withoutFormElement?: boolean;
   formConfig?: FormConfig;
   children: (formikProps: FormikProps<T>) => ReactElement;
+  onSubmit?: (values: T, formikHelpers: FormikHelpers<T>) => void | Promise<any>;
+}
+
+interface PersistedValuesProps<T> extends PersistValuesConfig<T> {
+  formikProps: FormikProps<T>;
 }
 
-const FormContent: FC<{
+function PersistedValues<T>(props: PersistedValuesProps<T>) {
+  const { formikProps, persistFunction, debounce } = props;
```

Comment:
Instead of requiring `formikProps` to be passed in, could we use the `useFormikContext` hook here?

```suggestion
  const { persistFunction, debounce } = props;
  const formikProps = useFormikContext();
```

Reply:
I'm wondering the same for our Input components, as well...